mantidproject/mantid
# Changed LoadSampleShape to rotate the sample based off of the rotation of the Goniometer

**Description of work.**
`LoadSampleShape` now gets the goniometer of the workspace its called on and uses that to work out the samples orientation, this was suggested by Pascal as a method of setting the orientation.


**To test:**
correct the WSFile and StlFile path for your External Data directories, then run the script
```
WSFile = '/home/sjenkins/Work/Build-1/ExternalData/Testing/Data/SystemTest/WISH/input/18_1/WISH00040503.raw'
StlFile = "/home/sjenkins/Work/Build-1/ExternalData/Testing/Data/UnitTest/cylinder.stl"

Load(Filename=WSFile, OutputWorkspace='ws',SpectrumList="100")
ConvertUnits(InputWorkspace="ws",Target="Wavelength", OutputWorkspace="ws1")
ConvertUnits(InputWorkspace="ws",Target="Wavelength", OutputWorkspace="ws2")
SetGoniometer(Workspace="ws1",Axis0="90,1,0,0,1")
LoadSampleShape(InputWorkspace="ws1",Filename=StlFile,Scale="cm",OutputWorkspace="ws1")
SetSampleMaterial(InputWorkspace="ws1",ChemicalFormula="V",SampleNumberDensity=0.1)
MonteCarloAbsorption(InputWorkspace="ws1",OutputWorkspace="rotatedAbsorb",NumberOfWavelengthPoints=50)
LoadSampleShape(InputWorkspace="ws2",Filename=StlFile,Scale="cm",OutputWorkspace="ws2")
SetSampleMaterial(InputWorkspace="ws2",ChemicalFormula="V",SampleNumberDensity=0.1)
MonteCarloAbsorption(InputWorkspace="ws2",OutputWorkspace="absorb",NumberOfWavelengthPoints=50)

```
plot absorb and rotatedAbsorb, rotatedAbsorb should be larger
Fixes #25499 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
